### PR TITLE
Change span to <p> in error message options

### PR DESCRIPTION
**Pull request by matteason**
In 4.0.0, [error messages were changed](https://github.com/alphagov/govuk-frontend/pull/2452) from `<span>`s to `<p>`s, but the macro documentation still references spans. This fixes those references

### CODE REVIEW

**Review by querkmachine — APPROVED** [review]
This all looks sensible to me. Just needs a second approval from someone on the team before we can merge.

**Review by owenatgov — APPROVED** [review]
This all looks sensible to me too. Thanks @matteason!

**Pull request merged**
